packages/sil-voting-app/src/routes/dashboard/components/Proposal.tsx

```diff
@@ -0,0 +1,191 @@
+  readonly status: 'Active' | 'Submitted' | 'Ended';
+}
+
+const getTotalVotes = (result: VoteResult): number => {
+  return result.yes + result.no + result.abstain;
+};
+
+const getWinner = (result: VoteResult): VoteWinner => {
+  const maxVotes = Math.max(result.yes, result.no);
+  let vote = 'Abstain';
+
+  if (maxVotes === result.yes) vote = 'Yes';
+  if (maxVotes === result.no) vote = 'No';
+  if (maxVotes === 0) vote = 'Abstain';
+
+  const percentage = Math.floor((maxVotes / getTotalVotes(result)) * 100);
+
+  return { vote, percentage };
+};
+
+const Proposal = (props: ProposalProps): JSX.Element => {
+  const [expanded, setExpanded] = useState(false);
+
+  const onClick = (): void => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <Block width="100%" display="flex" alignItems="center">
+      <Block flexGrow={1} margin={2}>
+        <Block display="flex" alignItems="center">
+          <Typography variant="h6">{elipsify(props.title, TITLE_MAX_LENGTH)}</Typography>
+          <Block marginLeft={2}>
+            <Typography variant="body1">{props.status}</Typography>
+          </Block>
+        </Block>
+        <Block display="flex" alignItems="center">
+          <Typography variant="body1">Author: {props.author}</Typography>
+          <Block marginLeft={2}>
+            <Typography variant="body1">Proposal ID: {props.id}</Typography>
+          </Block>
+          <Block marginLeft={2}>
+            <Typography variant="body1">
+              Created on {props.creationDate.toLocaleDateString('es-ES')}
+            </Typography>
+          </Block>
+        </Block>
+        {props.description.length < DESC_MAX_LENGTH && (
+          <Typography variant="body1">{props.description}</Typography>
+        )}
+        {props.description.length >= DESC_MAX_LENGTH && (
+          <React.Fragment>
+            {!expanded && (
+              <React.Fragment>
+                <Typography inline variant="body1">
+                  {elipsify(props.description, DESC_MAX_LENGTH)}{' '}
+                </Typography>
+                <Typography inline link onClick={onClick} variant="body1" weight="semibold">
+                  Read more
+                </Typography>
+              </React.Fragment>
+            )}
+            <Collapse in={expanded} timeout="auto" unmountOnExit>
+              <Typography inline variant="body1">
+                {props.description}{' '}
+              </Typography>
+              <Typography inline link onClick={onClick} variant="body1" weight="semibold">
+                Read less
+              </Typography>
+            </Collapse>
+          </React.Fragment>
+        )}
+        <Block display="flex" alignItems="center">
+          {props.status !== 'Ended' && (
+            <Typography variant="body1" weight="semibold">
+              Expires on {props.expiryDate.toLocaleDateString('es-ES')}
+            </Typography>
+          )}
+          {props.status === 'Ended' && (
```

Comment:
From line 106 to 168 (status conditions) should be refactored into a separate component Proposal/Status.tsx